File: LeetCode/14. 最长公共前缀.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
  let aStr = strs[0]
  for (let i = 1; i < strs.length; ++i) {
    let j = 0
    for (; j < aStr.length; j++) {
      if (strs[i][j] !== aStr[j]) break
    }
    aStr = aStr.substring(0, j)
    if (aStr === '') return aStr
  }
  return aStr
}

File: LeetCode/1-100/14. 最长公共前缀.java

class Solution {
    public String longestCommonPrefix(String[] strs) {
 String maxStr = strs[0];
        for (int i = 1; i < strs.length; i++) {
            String s = strs[i];
            int j = 0;
            for (; j < Math.min(s.length(), maxStr.length()); j++) {
                if (s.charAt(j) != maxStr.charAt(j)) break;
            }
            maxStr = maxStr.substring(0, j);
        }
        return maxStr;
    }
}